helm-v3/000-helm-v3.md

```diff
@@ -63,7 +63,10 @@ There are a few major (breaking) changes to the Helm CLI planned:
 
 - The `-n` flag will be remapped across the board to `--namespace`, not `-name`.
   This is for consistency with `kubectl`, which added `-n` after Helm 2 was released.
-- `helm install` will _require_ a name, unless `--generate-name` is specified. This
+- All actions will be organized under a noun to clarify what being acted on.
+  Some flags already work this way (e.g. `helm plugin add`) and some will need to be reorganized.
+  For example, `helm create` will become `helm chart create`.
```

Comment:
@jzelinskie Thanks for the comment. I'm guessing you're suggesting this for overall clarity for the end user experience.

Since we are trying to keep a consistent experience for existing Helm users I would expect aliases to be in place (e.g., `helm create` -> `helm chart create`) to ease any changes if this were adopted. This isn't a complete rewrite.

Out of curiosity, I went and looked at some other package managers to see what I could find.

* [Homebrew](https://brew.sh/) (for mac) has a simple set of commands similar to Helm today. For example, `brew create`, `brew install`, etc.
* [chocolately](https://chocolatey.org/) (for windows) also has a simple set of commands. For example, `choco new`, `choco install`, etc.
* [Snappy](https://snapcraft.io/) (for linux) is another example with simple commands. For example, `snap find`, `snap install`, etc.
* [Composer](https://getcomposer.org/) (for PHP) is yet another example of simple commands. They include `composer init` (like brew create or choco new), `composer install`, etc.
* [Yarn](https://yarnpkg.com/) (for JS) is, again, one with a simple command set including `yarn init`, `yarn install`, `yarn remove`, etc.

The more I look the more I see the common pattern is something simple.

Is there a compelling reason to add more structure to the commands?

Reply:
For those package managers listed, how many manipulate multiple objects without adding these prefixes to non-primary objects?

Let's look at brew:

```sh
$ brew
Example usage:
  brew search [TEXT|/REGEX/]
  brew info [FORMULA...]
  brew install FORMULA...
  brew update
  brew upgrade [FORMULA...]
  brew uninstall FORMULA...
  brew list [FORMULA...]

Troubleshooting:
  brew config
  brew doctor
  brew install --verbose --debug FORMULA

Contributing:
  brew create [URL [--no-fetch]]
  brew edit [FORMULA...]

Further help:
  brew commands
  brew help [COMMAND]
  man brew
  https://docs.brew.sh
```


I don't think this list is exhaustive because I know of commands like `brew link` that are not included here. However, it's clear that almost all the verbs all interact on one object type: Formula. Helm is a differs in that the user is frequently interacting with at least two types of objects: Charts and Releases. Even bigger is the divide that some commands work purely client-side and others interact with a server.

I think aliases for `helm create` -> `helm chart create` are a great compromise and achieves both simplicity and clarity.